packages/core/strapi/lib/services/entity-service/types/params/attributes.d.ts

```diff
@@ -18,6 +18,8 @@ export type GetNestedKeys<TSchemaUID extends Common.UID.Schema> = Exclude<
   GetNonFilterableKeys<TSchemaUID>
 >;
 
+export type ID = `${number}` | number;
```

Comment:
At first I thought this could be the wrong place to put this ID type, but since here it's being used as an attribute for the data parameters input for the entity service I think it's not a bad idea to have it here, even though we can have a generic ID type somewhere else (I don't know where tbh)

Reply:
Yeah we'll need to expose an ID type at a lower level so that it can be reused elsewhere. For now this is fine.